File: src/tests/javaobject/ExampleObject.java
/*=========================================================================
 * Copyright (c) 2010-2014 Pivotal Software, Inc. All Rights Reserved.
 * This product is protected by U.S. and international copyright
 * and intellectual property laws. Pivotal products are covered by
 * one or more patents listed at http://www.pivotal.io/patents.
 *=========================================================================
 */
//package org.apache.geode.cache.query.data;
package javaobject;

import java.util.*;
import java.io.*;
import org.apache.geode.*;
import org.apache.geode.cache.Declarable;

/**
* @brief Example 13.8 Implementing Complex Data Types Using the Java API.
*/
public class ExampleObject implements DataSerializable {
  private double double_field;
  private long long_field;
  private float float_field;
  private int int_field;
  private short short_field;
  private java.lang.String string_field;
  private Vector string_vector;
  static {
    Instantiator.register(new Instantiator(ExampleObject.class, (byte) 46)
    {
      public DataSerializable newInstance() {
        return new ExampleObject();
      }
    });
  }
  public ExampleObject( ) {
  this.double_field = 0.0D;
  this.long_field = 0L;
  this.float_field = 0.0F;
  this.int_field = 0;
  this.short_field = 0;
  this.string_field = null;
  this.string_vector = null;
  }
  public ExampleObject(int id) {
  this.int_field = id;
  this.string_field = String.valueOf(id);
  this.short_field = Short.parseShort(string_field);
  this.double_field = Double.parseDouble(string_field);
  this.float_field = Float.parseFloat(string_field);
  this.long_field = Long.parseLong(string_field);
  this.string_vector = new Vector();
  for (int i=0; i<3; i++) {
  this.string_vector.addElement(string_field);
  }
  }
  public ExampleObject(String id_str) {
  this.int_field = Integer.parseInt(id_str);
  this.string_field = id_str;
  this.short_field = Short.parseShort(string_field);
  this.double_field = Double.parseDouble(string_field);
  this.float_field = Float.parseFloat(string_field);
  this.long_field = Long.parseLong(string_field);
  this.string_vector = new Vector();
  for (int i=0; i<3; i++) {
    this.string_vector.addElement(string_field);
  }
  }
  public double getDouble_field( ) {
  return this.double_field;
  }
  public void setDouble_field( double double_field ) {
  this.double_field = double_field;
  }
  public long getLong_field( ) {
  return this.long_field;
  }
  public void setLong_field( long long_field ) {
  this.long_field = long_field;
  }
  public float getFloat_field( ) {
  return this.float_field;
  }
  public void setFloat_field( float float_field ) {
  this.float_field = float_field;
  }
  public int getInt_field( ) {
  return this.int_field;
  }
  public void setInt_field( int int_field ) {
  this.int_field = int_field;
  }
  public short getShort_field( ) {
  return this.short_field;
  }
  public void setShort_field( short short_field ) {
  this.short_field = short_field;
  }
  public java.lang.String getString_field( ) {
  return this.string_field;
  }
  public void setString_field( java.lang.String string_field ) {
  this.string_field = string_field;
  }
  public Vector getString_vector( ) {
  return this.string_vector;
  }
  public void setString_vector( Vector string_vector ) {
  this.string_vector = string_vector;
  }
  public void toData(DataOutput out) throws IOException
  {
    out.writeDouble(double_field);
    out.writeFloat(float_field);
    out.writeLong(long_field);
    out.writeInt(int_field);
    out.writeShort(short_field);
    out.writeUTF(string_field);
    out.writeInt(string_vector.size());
    for (int i = 0; i < string_vector.size(); i++)
    {
      out.writeUTF((String)string_vector.elementAt(i));
    }
  }
  public void fromData(DataInput in) throws IOException,
  ClassNotFoundException
  {
    this.double_field = in.readDouble();
    this.float_field = in.readFloat();
    this.long_field = in.readLong();
    this.int_field = in.readInt();
    this.short_field = in.readShort();
    this.string_field = in.readUTF();
    this.string_vector = new Vector();
    int size = in.readInt();
    for (int i = 0; i < size; i++)
    {
      String s = in.readUTF();
      string_vector.add(i, s);
    }
  }
  public boolean equals(ExampleObject o)
  {
    if (this.double_field != o.double_field) return false;
    if (this.float_field != o.float_field) return false;
    if (this.long_field != o.long_field) return false;
    if (this.int_field != o.int_field) return false;
    if (this.short_field != o.short_field) return false;
    if (!this.string_field.equals(o.string_field)) return false;
    if (!this.string_vector.equals(o.string_vector)) return false;
    return true;
  }
}

File: geode-client-native/examples/clicache/CacheRunner/ExampleObject.cs
/*=========================================================================
 * Copyright (c) 2002-2014 Pivotal Software, Inc. All Rights Reserved.
 * This product is protected by U.S. and international copyright
 * and intellectual property laws. Pivotal products are covered by
 * more patents listed at http://www.pivotal.io/patents.
 *========================================================================
 */

/*
 * @brief ExampleObject class for testing the put functionality for object. 
 */

using System;
namespace GemStone.GemFire.Cache.Examples
{
  public class ExampleObject
    : IGFSerializable
  {
    #region Private members
    private double double_field;
    private float  float_field;
    private long  long_field;
    private int  int_field;
    private short  short_field;
    private string  string_field;
      private System.Collections.ArrayList string_vector = new System.Collections.ArrayList();
    #endregion

    #region Public accessors
    public int Int_Field
    {
      get
      {
        return int_field;
      }
      set
      {
        int_field = value;
      }
    }
    public short Short_Field
    {
      get
      {
        return short_field;
      }
      set
      {
        short_field = value;
      }
    }
    public long Long_Field
    {
      get
      {
        return long_field;
      }
      set
      {
        long_field = value;
      }
    }
    public float Float_Field
    {
      get
      {
        return float_field;
      }
      set
      {
        float_field = value;
      }
    }
    public double Double_Field
    {
      get
      {
        return double_field;
      }
      set
      {
        double_field = value;
      }
    }
    public string String_Field
    {
      get
      {
        return string_field;
      }
      set
      {
        string_field = value;
      }
    }
    public System.Collections.ArrayList String_Vector
    {
      get
      {
        return string_vector;
      }
      set
      {
        string_vector = value;
      }
    }
    public override string ToString()
    {
        string buffer = "ExampleObject: "+int_field+"(int),"+string_field+"(str),";
        buffer += "[";
        for (int idx = 0; idx < string_vector.Count; idx++)
        {
            buffer += string_vector[idx];
        }
        buffer += "(string_vector)]";
        return buffer;
    }
    #endregion

    #region Constructors
    public ExampleObject()
    {
      double_field = (double)0.0;
      float_field = (float)0.0;
      long_field = 0;
      int_field = 0;
      short_field = 0;
      string_field = "";
      string_vector.Clear();
    }

    public ExampleObject(int id) {
      int_field = id;
      short_field = (Int16)id;
      long_field = (Int64)id;
      float_field = (float)id;
      double_field = (double)id;
      string_field = ""+id;
      string_vector.Clear();
      for (int i=0; i<3; i++) {
        string_vector.Add(string_field);
      }
    }
    public ExampleObject(string sValue) {
      int_field = Int32.Parse(sValue);
      long_field = Int64.Parse(sValue);
      short_field = Int16.Parse(sValue);
      double_field = (double)int_field;
      float_field = (float)int_field;
      string_field = sValue;
      string_vector.Clear();
      for (int i=0; i<3; i++) {
        string_vector.Add(sValue);
      }
    }
    #endregion

    #region IGFSerializable Members
    public IGFSerializable FromData(DataInput input)
    {
      double_field = input.ReadInt64();
      float_field = input.ReadFloat();
      long_field = input.ReadInt64();
      int_field = input.ReadInt32();
      short_field = input.ReadInt16();
      string_field = input.ReadUTF();
      int itemCount = input.ReadInt32();
      string_vector.Clear();
      for (int idx = 0; idx<itemCount; itemCount++) {
          string_vector.Add(input.ReadUTF());
      }
      return this;
    }

    public void ToData(DataOutput output)
    {
      output.WriteDouble( double_field );
      output.WriteFloat( float_field );
      output.WriteInt64( long_field );
      output.WriteInt32( int_field );
      output.WriteInt16( short_field );
      output.WriteUTF( string_field );
      int itemCount = string_vector.Count;
      output.WriteInt32( itemCount );
      for( int idx = 0; idx < itemCount; idx++ ) {
          string s = (string)string_vector[idx];
        output.WriteUTF( s );
      }
    }

    public UInt32 ObjectSize
    {
      get
      {
        UInt32 objectSize = 0;
        objectSize += (UInt32)sizeof(double);
        objectSize += (UInt32)sizeof(float);
        objectSize += (UInt32)sizeof(Int64);
        objectSize += (UInt32)sizeof(Int32);
        objectSize += (UInt32)sizeof(Int16);
        objectSize += (UInt32)(string_field == null ? 0 : sizeof(char) * string_field.Length);
        objectSize += (UInt32)sizeof(Int32);
        for (int idx = 0; idx<string_vector.Count; idx++) {
            string s = (string)string_vector[idx];
            objectSize += (UInt32)(string_vector[idx] == null ? 0 : sizeof(char) * s.Length);
        }
        return objectSize;
      }
    }

    public UInt32 ClassId
    {
      get
      {
        return 0x2e;
      }
    }
    #endregion

    public static IGFSerializable CreateDeserializable()
    {
      return new ExampleObject();
    }
  }

};



